client/main/contextmenu/contextmenubutton.js

```diff
@@ -0,0 +1,186 @@
+
+'use strict';
+
+const $ = require('jquery');
+const Backbone = require('backbone');
+const RibbonGroup = require('../ribbon/ribbongroup');
+
+const ActionHub = require('../actionhub');
+
+const ContextMenuButton = Backbone.View.extend({
+
+    /*
+    params
+    {
+        title:      //Title to be displayed
+        name:       //Button Id used in action event  REQUIRED!
+        tabName:    //Tab in which the button lives (namespace). Will add if not specified.
+        iconId:     //id of the icon to use. If null no icon will be used.
+        right:      //Is the button docked to the right? [default: false]
+        eventData:  //Data shared on events
+        useActionHub: //Should actionHub be called using the menu item name
+        enabled:    //initial enabled state
+        $el:        //jquery element. Will create if not defined.
+    }
+    */
+
+    initialize(params) {
```

Comment:
no biggie, but it is convention to call what you're calling `params`, `options`:

i.e.
https://electron.atom.io/docs/api/dialog/
https://nodejs.org/api/http.html#http_agent_getname_options